cli/flox/src/commands/mod.rs

```diff
@@ -423,12 +423,18 @@ impl UpdateNotification {
         cache_dir: impl AsRef<Path>,
     ) -> Result<Option<Self>, UpdateNotificationError> {
         let notification_file = cache_dir.as_ref().join(UPDATE_NOTIFICATION_FILE_NAME);
-        Self::check_for_update_inner(
-            notification_file,
-            Self::get_latest_version(),
-            UPDATE_NOTIFICATION_EXPIRY,
-        )
-        .await
+        // FLOX_SENTRY_ENV won't be set for development builds.
+        // Skip printing an update notification.
+        if let Some(ref sentry_env) = *FLOX_SENTRY_ENV {
+            Self::check_for_update_inner(
+                notification_file,
+                Self::get_latest_version(sentry_env),
```

Comment:
I didn't add a test because then I'd have to check sentry_env before calling `get_latest_version` in which case `get_latest_version` would have to be a function instead of a future. I can make that change if you think I should @zmitchell

Reply:
It’s probably fine